spanner/pdml.go

```diff
@@ -53,6 +53,10 @@ func (c *Client) partitionedUpdate(ctx context.Context, statement Statement, opt
 	}
 
 	sh, err := c.idleSessions.take(ctx)
+	// Mark isLongRunningTransaction to true, as the session in case of partitioned dml can be long-running
+	sh.mu.Lock()
```

Comment:
I recall that when a `Lock` was not taken here, there were some data-race conditions because the test cases were trying to read this property `isLongRunningTransaction` for validation.